### PR TITLE
Jb50035

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

**Comment by martyone** [comment]
Just the very last commit belongs to this PR

File: src/plugins/compilationdatabaseprojectmanager/compilationdatabaseutils.cpp [hunk 1]
```diff
@@ -148,9 +148,7 @@ void filteredFlags(const QString &fileName,
         }
 
         if (flag == "-o" || flag == "-MF" || flag == "-c" || flag == "-pedantic"
-                || flag.startsWith("-O") || flag.startsWith("-W") || flag.startsWith("-w")
-                || QString::compare(flag, "-fpic", Qt::CaseInsensitive) == 0
-                || QString::compare(flag, "-fpie", Qt::CaseInsensitive) == 0) {
+                || flag.startsWith("-O") || flag.startsWith("-W") || flag.startsWith("-w")) {
```
Comment (vigejolla) [inline src/plugins/compilationdatabaseprojectmanager/compilationdatabaseutils.cpp:151]: This is a bit strange, why were they filtered out in the first place? Have you tried upstreaming this change already?

Comment (martyone) [reply src/plugins/compilationdatabaseprojectmanager/compilationdatabaseutils.cpp:151]: Unfortunately that's unclear. It was added with one big commit enabling use with clang code model IIUC. Maybe something changed later on the clang code model that it makes not harm now. Not tried upstreaming yet.

**Comment by martyone** [comment]
Upstream bug filed https://bugreports.qt.io/browse/QTCREATORBUG-24106

**Pull request merged**